### PR TITLE
[SERVICES-1877] farm v2 events

### PR DESCRIPTION
**Pull request by claudiulataretu**
## Reasoning
- missing handlers for farm v2 events
  
## Proposed Changes
- added new methods to handle farm v2 events
- update rabbitmq handler to handle farm v2 events using new handlers

## How to test
- with `ENABLE_EVENTS_NOTIFIER=true` farm v2 events should be handled

### CODE REVIEW

**Review by cfaur09 — DISMISSED** [review]

File: src/modules/rabbitmq/rabbitmq.farm.handler.service.ts [hunk 1]
```diff
@@ -83,34 +76,65 @@ export class RabbitMQFarmHandlerService {
...
 
-    private async deleteCacheKeys() {
-        await this.pubSub.publish('deleteCacheKeys', this.invalidatedKeys);
-        this.invalidatedKeys = [];
+    async handleEnterFarmEventV2(rawEvent: RawEventType): Promise<void> {
+        const event = new EnterFarmEventV2(rawEvent);
+
+        const cacheKeys = await Promise.all([
+            this.farmSetterFactory
+                .useSetter(event.address)
```
Comment (dragos-rebegea) [inline src/modules/rabbitmq/rabbitmq.farm.handler.service.ts:106]: the list contains only 1 element, no need to use Promise.all here

File: src/modules/rabbitmq/rabbitmq.farm.handler.service.ts [hunk 2]
```diff
@@ -83,34 +76,65 @@ export class RabbitMQFarmHandlerService {
...
+
+        await this.deleteCacheKeys(cacheKeys);
+    }
+
+    async handleExitFarmEventV2(rawEvent: RawEventType): Promise<void> {
+        const event = new ExitFarmEventV2(rawEvent);
+
+        const cacheKeys = await Promise.all([
+            this.farmSetterFactory
+                .useSetter(event.address)
```
Comment (dragos-rebegea) [inline src/modules/rabbitmq/rabbitmq.farm.handler.service.ts:118]: the list contains only 1 element, no need to use Promise.all here

File: src/modules/rabbitmq/rabbitmq.farm.handler.service.ts [hunk 3]
```diff
@@ -83,34 +76,65 @@ export class RabbitMQFarmHandlerService {
...
+
+        await this.deleteCacheKeys(cacheKeys);
+    }
+
+    async handleClaimRewardsEventV2(rawEvent: RawEventType): Promise<void> {
+        const event = new ClaimRewardsEventV2(rawEvent);
+
+        const cacheKeys = await Promise.all([
+            this.farmSetterFactory
+                .useSetter(event.address)
```
Comment (dragos-rebegea) [inline src/modules/rabbitmq/rabbitmq.farm.handler.service.ts:130]: the list contains only 1 element, no need to use Promise.all here

File: src/modules/rabbitmq/rabbitmq.farm.handler.service.ts (hunk 2 above)
Comment (claudiulataretu) [reply src/modules/rabbitmq/rabbitmq.farm.handler.service.ts]: done

File: src/modules/rabbitmq/rabbitmq.farm.handler.service.ts (hunk 3 above)
Comment (claudiulataretu) [reply src/modules/rabbitmq/rabbitmq.farm.handler.service.ts]: done

**Review by dragos-rebegea — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Review by cfaur09 — APPROVED** [review]

**Pull request merged**